drivers/modem/modem_cellular.c

```diff
@@ -1261,6 +1277,39 @@ static inline int modem_cellular_csq_parse_rssi(uint8_t rssi, int16_t *value)
 	return 0;
 }
 
+MODEM_CHAT_SCRIPT_CMDS_DEFINE(get_signal_cesq_chat_script_cmds,
+			      MODEM_CHAT_SCRIPT_CMD_RESP("AT+CESQ", cesq_match),
+			      MODEM_CHAT_SCRIPT_CMD_RESP("", ok_match));
+
+MODEM_CHAT_SCRIPT_DEFINE(get_signal_cesq_chat_script, get_signal_cesq_chat_script_cmds,
+			 abort_matches, modem_cellular_chat_callback_handler, 2);
+
+/* AT+CESQ returns a response +CESQ: <rxlev>,<ber>,<rscp>,<ecn0>,<rsrq>,<rsrp> where:
+ * - rsrq is a integer from 0 to 34 whose values describes the Reference Signal Receive
+ *   Quality between -20 dB for 0 and -3 dB for 34 (0.5 dB steps), or unknown for 255
+ * - rsrp is an integer from 0 to 97 that describes the Reference Signal Receive Power
+ *   between -140 dBm for 0 and -44 dBm for 97 (1 dBm steps), or unknown for 255
+ */
+static inline int modem_cellular_cesq_parse_rsrp(uint8_t rsrp, int16_t *value)
+{
+	if (rsrp == 255) {
```

Comment:
could you add few definitions for these? based on the description above I imagine they could be something like 
```c
#define CESQ_RSII_UNKNOWN 255
#define CESQ_RSRP_TO_DB(v) (-140 + v)
#define CESQ_RSRQ_TO_DB(v) (-20 + v / 2)
```

then the macro somehow self-describe the input and output unit together with the formula

Reply:
Sure, however I don't think that necessarily makes it more clear. I think it's nice that the conversion functions are explicit about what's happening, so I don't have to go to a macro to find out. See what you think!

Reply:
I think it's better, last word to the maintainers though, @rerickson1?